test/mpi/Makefile.mtest

```diff
@@ -9,34 +9,6 @@
 ##
 ##   include $(top_srcdir)/Makefile.mtest
 ##
-## It provides a "testing" rule that uses runtests, cleanup for summary.xml, and
-## adds appropriate includes and linker flags for using the MTest framework.
-## The goal is to reduce the amount of boilerplate that needs to exist in each
-## Makefile.am
+## This makefile is for single-threaded tests.
 
-# AM_CPPFLAGS are used for C++ code as well
-AM_CPPFLAGS = -I$(top_builddir)/include -I$(top_srcdir)/include
-LDADD = $(top_builddir)/util/mtest.$(OBJEXT) $(top_builddir)/util/mtest_common.$(OBJEXT)
-
-# Add libdtpools support
-AM_CPPFLAGS += -I$(top_srcdir)/dtpools/include
-LDADD += $(top_builddir)/dtpools/src/libdtpools.la
-
-$(top_builddir)/util/mtest.$(OBJEXT): $(top_srcdir)/util/mtest.c
-	(cd $(top_builddir)/util && $(MAKE) mtest.$(OBJEXT))
-
-$(top_builddir)/util/mtest_common.$(OBJEXT): $(top_srcdir)/util/mtest_common.c
-	(cd $(top_builddir)/util && $(MAKE) mtest_common.$(OBJEXT))
-
-$(top_builddir)/dtpools/src/libdtpools.la:
-	(cd $(top_builddir)/dtpools && $(MAKE))
-
-TESTLIST ?= testlist,testlist.dtp,testlist.cvar
-SUMMARY_BASENAME ?= summary
-
-testing:
-	$(top_builddir)/runtests -srcdir=$(srcdir) -tests=$(TESTLIST) \
-		-mpiexec="${MPIEXEC}" -xmlfile=$(SUMMARY_BASENAME).xml \
-		-tapfile=$(SUMMARY_BASENAME).tap -junitfile=$(SUMMARY_BASENAME).junit.xml
-
-CLEANFILES = $(SUMMARY_BASENAME).xml $(SUMMARY_BASENAME).tap $(SUMMARY_BASENAME).junit.xml
+include $(top_srcdir)/Makefile_common.mtest
```

Comment:
How about rename this file into `Makefile_single.mtest`?

Reply:
It sounds nice. I will change so.

If so, how about creating only two object files, `mtest_single.o` and `mtest_threads.o` instead of the current `mtest.o`, `mtest_single.o`, and `mtest_threads.o`?

Reply:
Does that involve code duplication between `mtest_single.o` and `mtest_threads.o`?

Reply:
If it is the case, I will make `mtest.c` as a header (by making `mtest_common.h` or `mtest_base.h`), so no code duplication. It might look ugly (since functions are defined without `static` or `inline` in a header), though.

Reply:
> It might look ugly (since functions are defined without `static` or `inline` in a header), though.

It isn't worth it to save one object file, IMO.

Reply:
Thanks!

I got it. I will just add another commit that renames `Makefile.mtest` to `Makefile_single.mtest`. I will also check why tests failed and fix it.

Reply:
`Makefile.mtest` is renamed to `Makefile_singel.mtest.` in the latest version.